scripts/asm-installer/install_asm

```diff
@@ -839,13 +862,28 @@ ${CUSTOM_OVERLAY}
 EOF
   if [[ "${CA}" = "citadel" ]]; then
     ABS_OVERLAYS="${OPTIONS_DIRECTORY}/citadel-ca.yaml,${ABS_OVERLAYS}"
+  elif [[ "${CA}" = "private_ca" ]]; then
+    ABS_OVERLAYS="${OPTIONS_DIRECTORY}/private-ca.yaml,${ABS_OVERLAYS}"
   fi
   CUSTOM_OVERLAY="${ABS_OVERLAYS}"
 
   set_kpt_package_url
   WORKLOAD_POOL="${PROJECT_ID}.svc.id.goog"
 }
 
+validate_private_ca() {
```

Comment:
If these are just validating the arguments, this should be `validate_private_ca_args` and be called in `validate_args` instead of further down the stack. That way we can fail as early as possible if we know that the input is malformed.